src/Streamly/Streams/StreamK.hs

```diff
@@ -798,6 +868,22 @@ sequence m = go (toStream m)
             yieldk ma r = unStream (toStream $ ma |: go r) st stp sng yld
          in (unStream m1) (rstState st) stp single yieldk
 
+-------------------------------------------------------------------------------
+-- Inserting
+-------------------------------------------------------------------------------
+
+{-# INLINE intersperseM #-}
+intersperseM :: (IsStream t, MonadAsync m) => m a -> t m a -> t m a
+intersperseM a m = fromStream $ prependingStart (toStream m)
+    where
+    prependingStart m1 = Stream $ \st stp sng yld ->
+        let yieldk i x = unStream (return i |: go x) st stp sng yld
+         in unStream m1 st stp sng yieldk
+    go m2 = fromStream $ Stream $ \st stp sng yld ->
+        let single i = unStream (a |: yield i) st stp sng yld
+            yieldk i x = unStream (a |: return i |: go x) st stp sng yld
+         in unStream m2 st stp single yieldk
```

Comment:
Same as on line 881.

Reply:
We can add a `intersperseM` test in `transformCombineOpsCommon` to detect the illegal sharing of SVar. I think it will fail if we do not reset the state here.